### PR TITLE
Stop one of the machine stacking exploit

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
Stop This
come on
aaaaaa

## Why It's Good For The Game
No exploit! Bad!

## Changelog
:cl:
fix: fix machine stacking exploit using disposable tubes
/:cl:

### CODE REVIEW

File: code/game/machinery/_machinery.dm [hunk 1]
```diff
@@ -568,6 +568,10 @@
 
 /obj/proc/default_unfasten_wrench(mob/user, obj/item/I, time = 20) //try to unwrench an object in a WONDERFUL DYNAMIC WAY
 	if(!(flags_1 & NODECONSTRUCT_1) && I.tool_behaviour == TOOL_WRENCH)
+		var/turf/ground = get_turf(src)
+		if(!anchored && ground.is_blocked_turf(TRUE, TRUE, src))
```
Comment (Jared-Fogle) [inline code/game/machinery/_machinery.dm:572]: Change this to use named parameters--`TRUE, TRUE` is not obvious what it's doing.

File: code/game/machinery/_machinery.dm [hunk 2]
```diff
@@ -577,6 +581,9 @@
 		var/prev_anchored = anchored
 		//as long as we're the same anchored state and we're either on a floor or are anchored, toggle our anchored state
 		if(I.use_tool(src, user, time, extra_checks = CALLBACK(src, .proc/unfasten_wrench_check, prev_anchored, user)))
+			if(!anchored && ground.is_blocked_turf(TRUE, TRUE, src))//i know what you tryin to sneak in
```
Comment (Jared-Fogle) [inline code/game/machinery/_machinery.dm:584]: Here too.

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Just a quick code smell request, looks fine.

**Comment by LemonInTheDark** [comment]
The work is half done, thank you

File: code/game/turfs/turf.dm [hunk 3]
```diff
@@ -168,12 +168,14 @@ GLOBAL_LIST_EMPTY(station_turfs)
 /turf/proc/multiz_turf_new(turf/T, dir)
 	SEND_SIGNAL(src, COMSIG_TURF_MULTIZ_NEW, T, dir)
 
-///returns if the turf has something dense inside it. if exclude_mobs is true, skips dense mobs like fat yoshi.
-/turf/proc/is_blocked_turf(exclude_mobs)
+///returns if the turf has something dense inside it. if exclude_mobs is true, skips dense mobs like fat yoshi. if exclude_object is true, it will exclude the excluded_object you sent through
+/turf/proc/is_blocked_turf(exclude_mobs, exclude_object = FALSE, excluded_object)
```
Comment (LemonInTheDark) [inline code/game/turfs/turf.dm:172]: Why not just send in the excluded object, and if it exists exclude it?

Comment (Ghilker) [reply code/game/turfs/turf.dm:172]: Yer right, will change it in the next commit

**Pull request merged**